plugin/src/main/java/org/autorefactor/refactoring/rules/AndroidWakeLockRefactoring.java

```diff
@@ -0,0 +1,152 @@
+
+import org.autorefactor.refactoring.ASTBuilder;
+import org.autorefactor.refactoring.FinderVisitor;
+import org.autorefactor.refactoring.Refactorings;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.SingleVariableDeclaration;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+
+/** See {@link #getDescription()} method. */
+public class AndroidWakeLockRefactoring extends AbstractRefactoringRule {
+
+    @Override
+    public String getDescription() {
+        return "Failing to release a wakelock properly can keep the Android device in a high "
+            + "power mode, which reduces battery life. There are several causes of this, such "
+            + "as releasing the wake lock in onDestroy() instead of in onPause(), failing to "
+            + "call release() in all possible code paths after an acquire(), and so on.";
+    }
+
+    @Override
+    public String getName() {
+        return "Android WakeLock";
+    }
+
+    @Override
+    public boolean visit(MethodInvocation node) {
+        if (isMethod(node, "android.os.PowerManager.WakeLock", "release")) {
+            // check whether it is being called in onDestroy
+            MethodDeclaration enclosingMethod = getAncestor(node, MethodDeclaration.class);
+            if (isMethod(enclosingMethod.resolveBinding(), "android.app.Activity", "onDestroy")) {
+                final Refactorings r = this.ctx.getRefactorings();
+                TypeDeclaration typeDeclaration = getAncestor(enclosingMethod, TypeDeclaration.class);
+                MethodDeclaration onPauseMethod = findMethod(typeDeclaration, "onPause");
+                if (onPauseMethod != null && node.getParent().getNodeType() == ASTNode.EXPRESSION_STATEMENT) {
+                    r.remove(node.getParent());
+                    r.insertLast(createWakelockReleaseNode(node), onPauseMethod);
+                    return DO_NOT_VISIT_SUBTREE;
+                }
+                /* If it reaches this part of the code, it
+                 * means it did not find onPause method in the class.
+                 */
+                // add onPause declaration to the Activity
+                r.insertAfter(createOnPauseMethodDeclaration(), enclosingMethod);
+                return DO_NOT_VISIT_SUBTREE;
+            }
+        } else if (isMethod(node, "android.os.PowerManager.WakeLock", "acquire")) {
+            final Refactorings r = this.ctx.getRefactorings();
+            TypeDeclaration typeDeclaration = getAncestor(node, TypeDeclaration.class);
+            ReleasePresenceChecker releasePresenceChecker = new ReleasePresenceChecker();
+            if (!releasePresenceChecker.findOrDefault(typeDeclaration, false)) {
+                Statement releaseNode = createWakelockReleaseNode(node);
```

Comment:
Please reduce scope of `releaseNode` local variable. :)

i.e.e move it in the if statement just before where it is used.